src/extension/wps/wps-core/src/main/java/org/geoserver/wps/ChainedProcessListenerNotifier.java

```diff
@@ -0,0 +1,50 @@
+/* (c) 2022 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wps;
+
+import java.util.List;
+import org.geoserver.wps.executor.ProcessManager;
+
+/** A notifier to fire events when processes are (un)chained.
+ * 
+ *  @author etj (Emanuele Tajariol @ GeoSolutions)
+ */
+public class ChainedProcessListenerNotifier {
```

Comment:
Please avoid public visibility for classes unless they need to be visible in other packages to work. As a support class, it could be placed in the executor package and be done with it, is there anything else that needs to use it?